src/metabase/util/malli/fn.clj

```diff
@@ -344,7 +378,10 @@
   problem for another day. The passed function name comes back from [[mc/parse]] as `:name` if we want to attempt to
   fix this later."
   [& fn-tail]
-  (let [error-context (if (symbol? (first fn-tail))
-                        {:fn-name (list 'quote (first fn-tail))}
-                        {})]
-    (instrumented-fn-form error-context (parse-fn-tail fn-tail))))
+  (let [parsed (parse-fn-tail fn-tail)]
+    `(if (#'*skip-ns-decision-fn*)
```

Comment:
If I'm reading this right, this is compiling the `if` into the code where the function is being constructed. That doesn't seem ideal... but there doesn't seem to be a way to check at macro expansion time.

I suppose it's not so bad if it's ultimately a top-level thing like a `mu/defn`.

Reply:
Yeah, I didn't start with this approach, but there doesn't seem to be a way to check at macro-expansion time. 

Pointing out the obvious, but the `if` is evaluated at `mu.fn/fn` compile-time, so any extra time to run it will occur during fn creation, and not fn execution.

Reply:
I think we have everything we need at macro-expansion time right? The only input is the namespace which is known at compile time. I think that condition can be outside of emitted code.

Reply:
I couldn't figure out how to get it working with re-binding or alter-var-rooting or swap!ing the state required to test it without emitting the if. Maybe I missed something there?

Reply:
I got the emission (and altering behavior in tests) working properly, with help from @dpsutton and @noahmoss 😎.